pkg/cmd/roachtest/tests/backup_restore_roundtrip.go

```diff
@@ -416,3 +436,128 @@ func createDriversForBackupRestore(
 	}
 	return d, runBackgroundWorkload, tables, nil
 }
+
+func testOnlineRestoreRecovery(ctx context.Context, t test.Test, c cluster.Cluster) {
+	testRNG, seed := randutil.NewLockedPseudoRand()
+	t.L().Printf("random seed: %d", seed)
+
+	envOption := install.EnvOption([]string{
+		"COCKROACH_MIN_RANGE_MAX_BYTES=1",
+	})
+
+	c.Start(ctx, t.L(), roachtestutil.MaybeUseMemoryBudget(t, 50), install.MakeClusterSettings(envOption), c.CRDBNodes())
+	m := c.NewMonitor(ctx, c.CRDBNodes())
+	const jobStatusWait = time.Minute * 5
+
+	m.Go(func(ctx context.Context) error {
+		testUtils, err := setupBackupRestoreTestUtils(
+			ctx, t, c, m, testRNG,
+			withOnlineRestore(true), withCompaction(true),
+		)
+		if err != nil {
+			return err
+		}
+		defer testUtils.CloseConnections()
+
+		dbs := []string{"bank", "tpcc", schemaChangeDB}
+		d, runBackgroundWorkload, _, err := createDriversForBackupRestore(
+			ctx, t, c, m, testRNG, testUtils, dbs,
+		)
+		if err != nil {
+			return err
+		}
+
+		stopBackgroundCommands, err := runBackgroundWorkload()
+		if err != nil {
+			return err
+		}
+		defer stopBackgroundCommands()
+
+		dbConn := c.Conn(ctx, t.L(), c.CRDBNodes().RandNode()[0])
+		defer dbConn.Close()
+		if _, err := dbConn.ExecContext(
+			ctx, "SET CLUSTER SETTING jobs.debug.pausepoints = 'restore.before_download'",
```

Comment:
not sure why you need to set this pause point

Reply:
Explained above

Reply:
nit: set the pausepoint before we run the restore? or do we set it up here because the setting needs to be set for full cluster restores? regardless, i'd document why we do this.

Reply:
Oh sure, I'll move it. And yeah I'll add a comment